### ISSUE TITLE
policy_id in policy resource always changes to null on tf apply

### ISSUE THREAD

**Issue by solalraveh** [PaloAltoNetworks/terraform-provider-prismacloud#19]
### Is your feature request related to a problem?

When provisioning a policy resource, adding all possible fields to its compliance metadata, there's one field that can't be populated by referencing other resources (but only explicit text).
This field is `policy_id`, is also used in the REST API and isn't mandatory nor critical in compliance metadata.
Since it's a field, every time a `tf apply` is performed, even if there's no change to the file, then it considers a change to the policy's compliance metadata settings, setting `policy_id` from the actual ID to null.
Example:
```
    compliance_metadata {
      compliance_id     = prismacloud_compliance_standard_requirement_section.CSRS_PANW_TEAM1_EC2.csrs_id
      custom_assigned = true
      requirement_id = prismacloud_compliance_standard_requirement.CSR_PANW_TEAM1.requirement_id
      requirement_name = prismacloud_compliance_standard_requirement.CSR_PANW_TEAM1.name
      section_id = prismacloud_compliance_standard_requirement_section.CSRS_PANW_TEAM1_EC2.section_id
      standard_name = prismacloud_compliance_standard.CS_PANW.name
      section_description = prismacloud_compliance_standard_requirement_section.CSRS_PANW_TEAM1_EC2.description
      standard_description = prismacloud_compliance_standard.CS_PANW.description
      #only mandatory parameter is compliance_id which corresponds to the section ID (not requirement and not compliance. section ID will identify those 2 automatically).
    }
```

Then running apply each time (no change at all) :
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # prismacloud_policy.this will be updated in-place
  ~ resource "prismacloud_policy" "this" {
        cloud_type               = "aws"
        created_by               = "yada yadi"
        created_on               = 1612861894487
        deleted                  = false
        description              = "yada"
        enabled                  = true
        id                       = "548cdbc2-cce9-4cd8-a66d-346b71f287c5"
        labels                   = []
        last_modified_by         = "yada yadi"
        last_modified_on         = 1612862636887
        name                     = "test aws EC2s"
        open_alerts_count        = 0
        overridden               = false
        owner                    = "yada yadu"
        policy_id                = "548cdbc2-cce9-4cd8-a66d-346b71f287c5"
        policy_mode              = "custom"
        policy_type              = "config"
        recommendation           = "yada"
        remediable               = false
        restrict_alert_dismissal = false
        rule_last_modified_on    = 1612861894487
        severity                 = "medium"
        system_default           = false

      ~ compliance_metadata {
            compliance_id        = "4672318a-3617-4e37-9606-e882c1ed5e9a"
            custom_assigned      = true
          - policy_id            = "548cdbc2-cce9-4cd8-a66d-346b71f287c5" -> null
            requirement_id       = "PANW Building Blocks"
            requirement_name     = "PANW - Building Blocks"
            section_description  = "Requirement Section for EC2"
            section_id           = "EC2"
            standard_description = "Compliance Standards PANW"
            standard_name        = "PANW"
        }

        rule {
            criteria   = "35f38c75-4472-436b-b085-f2be87fde644"
            name       = "test aws EC2s"
            parameters = {
                "savedSearch" = "true"
                "withIac"     = "false"
            }
            rule_type  = "Config"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

The key part is:
```
          - policy_id            = "548cdbc2-cce9-4cd8-a66d-346b71f287c5" -> null
```

### Describe the solution you'd like

Since there's no way to reference the policy ID from the compliance resource, nor policy resource (TF identifies it as a resource referencing itself and not allowing it), nor by declaring a policy datasource, then only explictly entering a text here will work. But that means one has no way of knowing what to put in the text, as the policy ID will only be known after the first `tf apply`

Since this `policy_id` fields under compliance_metadata in policy resource is: not mandatory, not needed, not playing any critical role and is auto-updated to the correct value even if not called out, then I suggest removing the print option back to TF when a policy is touched and not floating the `policy_id` under compliance metadata back to TF, so it won't be aware of this field and won't think it's changing to null every time.
It can still be used if someone wants to, but just to not float it back to TF on apply.

### Describe alternatives you've considered

Tried filling in the value by calling out the policy_id from its own resource (policy), also calling out from the output of the policy datasource and from compliance resource. 
This won't work with the example:
```
Error: "name": one of `name,policy_id` must be specified

  on prisma-cloud-policy-compliance.tf line 91, in data "prismacloud_policy" "example":
  91: data "prismacloud_policy" "example" {
```

or
```
Error: Incorrect attribute value type

  on prisma-cloud-policy-compliance.tf line 84, in resource "prismacloud_policy" "this":
  84:       policy_id = prismacloud_compliance_standard_requirement_section.CSRS_PANW_TEAM1_EC2.associated_policy_ids
    |----------------
    | prismacloud_compliance_standard_requirement_section.CSRS_PANW_TEAM1_EC2.associated_policy_ids is list of string with 1 element

Inappropriate value for attribute "policy_id": string required.
```

or
```
Error: Self-referential block

  on prisma-cloud-policy-compliance.tf line 84, in resource "prismacloud_policy" "this":
  84:       policy_id = prismacloud_policy.this.id

Configuration for prismacloud_policy.this may not refer to itself.


Error: Self-referential block

  on prisma-cloud-policy-compliance.tf line 84, in resource "prismacloud_policy" "this":
  84:       policy_id = prismacloud_policy.this.id

Configuration for prismacloud_policy.this may not refer to itself.
```

### Additional context

The file in question (Policy Resource) is https://github.com/PaloAltoNetworks/terraform-provider-prismacloud/blob/master/prismacloud/resource_policy.go

With an ever increasing estate, every tf plan with dozens of policies is causing parsing issues and a lot of print outs with modifications that aren't really happening, simply because policy_id can't be declared programmatically. 

CC @n-schilling

**Comment by solalraveh** [comment]
@ebeuerle I'll chat to you offline and TIA!

**Comment by tomsucho** [comment]
Hey! is there a workaround/solution for this?

**Comment by ebeuerle** [comment]
I am working to get this fixed this week.

**Comment by ebeuerle** [comment]
Fixed in next version.

**Comment by ebeuerle** [comment]
Fixed as of 1.1.2.
